app/queries/bp_measures_query.rb

```diff
@@ -0,0 +1,22 @@
+class BPMeasuresQuery
+  def count(region, period_type, diagnosis: :hypertension, group_by: nil)
+    query = BloodPressure
+      .joins(:patient).merge(Patient.with_hypertension)
+      .group_by_period(period_type, :recorded_at, {format: Period.formatter(period_type)})
+      .where(facility: region.facilities)
+    if group_by.present?
+      results = query.group(group_by).count
+      sum_groups_by_period(results)
+    else
+      query.count
+    end
+  end
+
+  def sum_groups_by_period(result)
```

Comment:
We're flipping the data structure so the period is the first key, and the other grouping is the second? Should we remove the word `sum` from this method name?

Reply:
Right, this is more inline w/ all the other data the Repo returns.  The structure is:

```
region1 => { period1 => xxx, period2 => yyy },
region2 => { period1 => xxx, period2 => yyy },
```

So for data that has an additional grouping, I'm going w/ the convention of:

```
region1 => { period1 => {group1 => x, group2 => y}, period2 => {group2 => z} },
region2 => { etc... }
```

The goal is that all data from Repository will follow this same sort of shape to stay consistent.